pkg/services/playlist/playlistimpl/sqlx_store.go

```diff
@@ -0,0 +1,246 @@
+package playlistimpl
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/playlist"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/jmoiron/sqlx"
+)
+
+type sqlxStore struct {
+	sqlxdb  *sqlx.DB
+	dialect migrator.Dialect
+}
+
+func (s *sqlxStore) insertWithReturningId(ctx context.Context, tx *sqlx.Tx, p *playlist.Playlist) error {
+	if s.dialect.DriverName() == "postgres" {
+		query := fmt.Sprintf("INSERT INTO playlist (name, %s, org_id, uid) VALUES (?, ?, ?, ?) RETURNING id", s.dialect.Quote("interval"))
```

Comment:
Look good, but this block would be the only concerning part of the PR to me.
Sure, SQL has delimited identifiers that must be quoted (whitespace, case-sensitive, matching language keywords like this one, etc).
The common way to quote them would be double quotes, i.e. `"interval"`.
If you write it like this directly in the query - it should work for SQLite and posteres already.
MySQL is a special case, they use backticks by default.
I see two options here:
* Create a wrapper over sqlx that does this substitution inside questies based on the driver
* Use `SET SQL_MODE=ANSI_QUOTES;` for all mysql connections to enforce the consistent behaviour between the databases. Just in case, MSSql also has an option to use `"` by default.

Double quote usage is defined as a part of the SQL92 standard and I think we should use that inside queries, trying not to come up with unique queries for each dialect whenever it's possible.